pyat/at/collective/beam_loading.py

```diff
@@ -139,12 +145,15 @@ def __init__(self, family_name: str, length: float, voltage: float,
                 (default) uses the phasor method, BLMode.WAKE uses the wake
                 function. For high Q resonator, the phasor method should be
                 used
+            cavitymode (CavityMode):  Define cavity as active (default) or passive 
         Returns:
             bl_elem (Element): beam loading element
         """
         kwargs.setdefault('PassMethod', self.default_pass[True])
         assert isinstance(mode, BLMode), \
             'Beam loading mode has to be an instance of BLMode'
+        assert isinstance(cavitymode, CavityMode), \
+            'Cavity mode has to be an instance of CavityMode'
```

Comment:
As a general rule, you should avoid using `assert` for validity checks: `assert` statements are only executed when python is in debug mode, but  not in optimisation mode (python -O). Debug mode is currently the default, but it may change.
Instead, raise exceptions (TypeError in this case)